src/bench/bench.h

```diff
@@ -31,50 +33,97 @@ static void CODE_TO_TIME(benchmark::State& state)
     ... do any cleanup needed...
 }
 
-BENCHMARK(CODE_TO_TIME);
+// default to running benchmark for 5000 iterations
+BENCHMARK(CODE_TO_TIME, 5000);
 
  */
- 
+
 namespace benchmark {
+class Printer;
 
-    class State {
-        std::string name;
-        double maxElapsed;
-        double beginTime;
-        double lastTime, minTime, maxTime;
-        uint64_t count;
-        uint64_t countMask;
-        uint64_t beginCycles;
-        uint64_t lastCycles;
-        uint64_t minCycles;
-        uint64_t maxCycles;
-    public:
-        State(std::string _name, double _maxElapsed) : name(_name), maxElapsed(_maxElapsed), count(0) {
-            minTime = std::numeric_limits<double>::max();
-            maxTime = std::numeric_limits<double>::min();
-            minCycles = std::numeric_limits<uint64_t>::max();
-            maxCycles = std::numeric_limits<uint64_t>::min();
-            countMask = 1;
-        }
-        bool KeepRunning();
-    };
+class State
+{
+public:
+    std::string m_name;
+    uint64_t m_num_iters_left;
+    const uint64_t m_num_iters;
+    const uint64_t m_num_evals;
+    std::vector<double> m_elapsed_results;
+    Printer& m_printer;
 
-    typedef std::function<void(State&)> BenchFunction;
+    std::chrono::time_point<std::chrono::high_resolution_clock> m_start_time;
 
-    class BenchRunner
+    bool UpdateTimer(std::chrono::time_point<std::chrono::high_resolution_clock> finish_time);
+    void PrintResults() const;
```

Comment:
This function is neither defined, nor used.

Reply:
thanks, I've removed it